Add qualifier output and model family. More detailed data_paths.

mkozlowski-uncharted opened the pull request:
Added `qualifier_outputs` to the indicator and model schemas.
Added `family_name` to the indicator and model schemas.
In the model run schema, `data_paths` is now an object with two array fields: `number` and `str`. The URL paths are now placed in these arrays depending on whether the `value` column in the data file is a `string` or a `number` type

mkozlowski-uncharted commented:
@brandomr @jryu01 @amostafa-uncharted

brandomr commented:
@mkozlowski-uncharted the only issue I have is with the `data_paths`. This would require us to retrofit all existing metadata. 

The other changes would (e.g. `family_name`) would not require retrofitting since `family_name` is optional and the old indicators just won't have `qualifiers`.

We _could_ do this retrofitting if necessary, but it would be nice to avoid it.

brandomr commented:
Also, `indicators` will have the same potential issue as model outputs where there could be features that have `str` and features that are `float`. We'd end up with 2 parquet files for the indicator. However this is not reflected in the `indicator` schema.

Therefore I'd again suggest we leave `data_paths` as is for the time being and address it later if need be.

mkozlowski-uncharted commented:
@brandomr Yeah, reindexing all the current documents would be a pain. I've removed those changes for now, we can revisit this later.
FYI: The `family_name` **is** required in the model schema, I hope that's ok. Since there's only 2 models right now adding that should be fairly easy.

@travis-jat Thanks for catching that! Turns out I wrote the definition for that object but didn't actually add the property. Fixed now. I used `"qualifier_outputs"` to be consistent with `outputs` and `parameters`. Sorry for the confusion.
